index.js

```diff
@@ -0,0 +1,279 @@
+        var lcount = 0;
+        var rcount = 0;
+        var len = str.length;
+        var func = "";
+        for (i=0; i<len; i++) {
+            if (str[i] = "(") {
+                lcount += 1;
+            }
+            if (str[i] = ")") {
+                rcount += 1;
+            }
+    
+            if (((/[a-z]/).test(str[i]))) {
+                if (!(/[a-zA-Z]/).test(str[i-1]) && !(/[a-zA-Z]/).test(str[i-1])) {
+                    if(!vars.includes(str[i])) {
+                        return false;
+                    }
+                }
+            }
+            if ((/[A-Z]/).test(str[i])) {
+                func = func + str[i];
+            }
+            else {
+                if (funs.includes(func) || func == "") {
+                    func = "";
+                }
+                else {
+                    return false;
+                }
+            }
+        }
+    
+        return (lcount == rcount);
+    }
+    
+    /**
+     * Identifies convertable functions in expression and converts them recursively.
+     * @param {string} str - inputted normal syntax expression
+     * @returns expression with converted functions
+     */
+    function convert(str) {
+        var count = 0;
+        if (str.includes("^")) {
+            var i = str.indexOf("^");
+            var base = lfind(str, i);
+            var power = rfind(str, i);
+            str = str.slice(0, i-base.length) + "(" + base + ".power(" + power + ")" + ")" + str.slice(i+power.length+1);
+            count += 1;
+        }
+        if (str.includes("CEILING")) {
+            str = funcappend(str, "CEILING");
+            count += 1;
+        }
+        if (str.includes("FLOOR")) {
+            str = funcappend(str, "FLOOR");
+            count += 1;
+        }
+        if (str.includes("ABS")) {
+            str = funcappend(str, "ABS");
+            count += 1;
+        }
+        if (str.includes("SQRT")) {
+            str = funcappend(str, "SQRT");
+            count += 1;
+        }
+        if (str.includes("TRUNCATE")) {
+            str = funcappend(str, "TRUNCATE");
+            count += 1;
+        }
+        if (str.includes("EXP")) {
+            str = funcappend(str, "EXP");
+            count += 1;
+        }
+        if (str.includes("LN")) {
+            str = funcappend(str, "LN");
+            count += 1;
+        }
+        if (str.includes("LOG")) {
+            str = logappend(str);
+            count += 1;
+        }
+        if (count != 0) {
+            return convert(str);
+        }
+        else {
+            return str;
+        }
+    }
+    
+    /**
+     * Identifies functions and appends them in fhirpath form 
+     * @param {string} str - inputted normal syntax expression
+     * @param {string} func - function in inputted normal syntax expression
+     * @returns expression with converted function
+     */
+    function funcappend(str, func){
+        var i = str.indexOf(func);
```

Comment:
I believe indexOf is case sensitive but the calling function uses the case insensitive .includes method; what's the reason behind that?

Reply:
I'm just using indexOf here since it needs the index the function. The calling function uses includes to identify if there are any nonconverted parts of the function. Additionally, the parser assumes all functions are written in caps.